cmds/core/touch/touch.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+package main
+
+import (
+	"flag"
+	"log"
+	"os"
+	"time"
+)
+
+type params struct {
+	time         time.Time
+	access       bool
+	modification bool
+	create       bool
+}
+
+type cmd struct {
+	params
+	args []string
+}
+
+func command(p params, args ...string) *cmd {
+	return &cmd{
+		args:   args,
+		params: p,
+	}
+}
+
+func parseParams(dateTime string, access, modification, create bool) (params, error) {
+	flag.Parse()
+	t := time.Now()
+	if dateTime != "" {
+		var err error
+		t, err = time.Parse(time.RFC3339, dateTime)
+		if err != nil {
+			return params{}, err
+		}
+	}
+	return params{
+		access:       access || (!access && !modification),
+		modification: modification || (!access && !modification),
+		create:       create,
+		time:         t,
+	}, nil
+}
+
+func (c *cmd) run() error {
+	for _, arg := range c.args {
```

Comment:
ronsexcllentmbp:tmp rminnich$ touch a /etc/hosts b
touch: /etc/hosts: Permission denied
ronsexcllentmbp:tmp rminnich$ ls -l a b
-rw-r--r--  1 rminnich  wheel  0 Dec 11 08:33 a
-rw-r--r--  1 rminnich  wheel  0 Dec 11 08:33 b

note that it does not stop on first errors. I suspect you could use errors.Join, and run through this whole list, and return the cumulative error.

Reply:
Thx, nice catch. I think we still need to wait for errors.Join until next Go upgrade. I just added stderr for logging and make a todo not to change it in future.